segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```
@@ -0,0 +1,266 @@
+
+    @Before
+    public void setUp() throws Exception {
+
+        URI uri = URI.create(endpoint);
+        Properties properties = new Properties();
+        properties.setProperty("s3proxy.authorization", "none");
+        properties.setProperty("s3proxy.endpoint", endpoint);
+        properties.setProperty("jclouds.provider", "filesystem");
+        properties.setProperty("jclouds.filesystem.basedir", "/tmp/s3proxy");
+
+        ContextBuilder builder = ContextBuilder
+                .newBuilder("filesystem")
+                .credentials("x", "x")
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .overrides(properties);
+        BlobStoreContext context = builder.build(BlobStoreContext.class);
+        BlobStore blobStore = context.getBlobStore();
+
+        s3Proxy = S3Proxy.builder().awsAuthentication( AuthenticationType.AWS_V2_OR_V4, "x", "x")
+                .endpoint(uri)
+                .keyStore("", "")
+                .blobStore(blobStore)
+                .ignoreUnknownHeaders(true)
+                .build();
+
+        s3Proxy.start();
+
+        this.adapterConfig = ExtendedS3StorageConfig.builder()
+                                                    .with(ExtendedS3StorageConfig.BUCKET, BUCKET_NAME)
+                                                    .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                    .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                    .with(ExtendedS3StorageConfig.ROOT, "test")
+                                                    .with(ExtendedS3StorageConfig.URI, endpoint)
+                                                    .build();
+        createStorage();
+
+        try {
+            client.createBucket(BUCKET_NAME);
+        } catch (S3Exception e) {
+            if (!e.getErrorCode().equals("BucketAlreadyOwnedByYou")) {
```

Is it possible that in the case the bucket has been created previously that it is dirty and could affect the test results? I'm wondering if we should fail in this case rather than ignore.